drake/solvers/MathematicalProgram.cpp

```diff
@@ -31,41 +31,41 @@ class MathematicalProgram : public MathematicalProgramInterface {
    virtual MathematicalProgramInterface* addComplementarityConstraint() { return new
    MathematicalProgram; };
 */
-  virtual MathematicalProgramInterface* addGenericObjective() override {
+  virtual MathematicalProgramInterface* add_generic_objective() override {
```

Comment:
Convince me that "you normally wouldn't bother caching its return value when calling it in a loop"?